crates/aptos/src/move_tool/disassembler.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{CliCommand, CliError, CliTypedResult};
+use anyhow::Context;
+use async_trait::async_trait;
+use clap::Parser;
+use move_binary_format::{
+    binary_views::BinaryIndexedView, file_format::CompiledScript, CompiledModule,
+};
+use move_bytecode_source_map::{mapping::SourceMapping, utils::source_map_from_file};
+use move_command_line_common::files::{
+    MOVE_COMPILED_EXTENSION, MOVE_EXTENSION, SOURCE_MAP_EXTENSION,
+};
+use move_coverage::coverage_map::CoverageMap;
+use move_disassembler::disassembler::{Disassembler, DisassemblerOptions};
+use move_ir_types::location::Spanned;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+/// Disassemble the Move bytecode pointed to
+#[derive(Debug, Parser)]
+pub struct Disassemble {
+    /// Skip printing of private functions.
+    #[clap(long)]
+    pub skip_private: bool,
+
+    /// Do not print the disassembled bytecodes of each function.
+    #[clap(long)]
+    pub skip_code: bool,
+
+    /// Do not print locals of each function.
+    #[clap(long)]
+    pub skip_locals: bool,
+
+    /// Do not print the basic blocks of each function.
+    #[clap(long)]
+    pub skip_basic_blocks: bool,
+
+    /// Treat input file as a script (default is to treat file as a module)
+    #[clap(long)]
+    pub is_script: bool,
+
+    /// The path to the bytecode file to disassemble; let's call it file.mv. We assume that two
+    /// other files reside under the same directory: a source map file.mvsm (possibly) and the Move
+    /// source code file.move.
+    #[clap(long)]
+    pub bytecode_path: PathBuf,
+
+    /// (Optional) Path to a coverage file for the VM in order to print trace information in the
+    /// disassembled output.
+    #[clap(long)]
+    pub code_coverage_path: Option<PathBuf>,
+}
+
+#[async_trait]
+impl CliCommand<String> for Disassemble {
+    fn command_name(&self) -> &'static str {
+        "Disassemble"
+    }
+
+    async fn execute(self) -> CliTypedResult<String> {
+        let move_extension = MOVE_EXTENSION;
+        let mv_bytecode_extension = MOVE_COMPILED_EXTENSION;
+        let source_map_extension = SOURCE_MAP_EXTENSION;
+
+        let source_path = Path::new(&self.bytecode_path);
+        let extension = source_path
+            .extension()
+            .context("Missing file extension for bytecode file")?;
+        if extension != mv_bytecode_extension {
+            return Err(CliError::UnexpectedError(format!(
+                "Bad source file extension {:?}; expected {}",
+                extension, mv_bytecode_extension
+            )));
+        }
+
+        let bytecode_bytes =
+            fs::read(&self.bytecode_path).context("Unable to read bytecode file")?;
+
+        let source_path = Path::new(&self.bytecode_path).with_extension(move_extension);
+        let source = fs::read_to_string(&source_path).ok();
```

Comment:
This will ensure the errors are handled if the source isn't a valid UTF-8 string.
```suggestion
        let source = String::from_utf8(read_from_file(old_config_file.as_path())?)?
```

Reply:
I think this will return a String, but the original one returns an Optional String?